src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -177,9 +172,15 @@ void Qgs3DMapCanvas::saveAsImage( const QString fileName, const QString fileForm
 {
   if ( !fileName.isEmpty() )
   {
-    mCaptureFileName = fileName;
-    mCaptureFileFormat = fileFormat;
-    mEngine->requestCaptureImage();
+    QgsOffscreen3DEngine engine;
+    engine.setSize( size() );
+    Qgs3DMapScene scene( *mMap, &engine );
+    engine.setRootEntity( &scene );
+    QgsCameraController *cameraController = mScene->cameraController();
+    scene.cameraController()->setLookingAtPoint( cameraController->lookingAtPoint(), cameraController->distance(), cameraController->pitch(), cameraController->yaw() );
+    QImage img = Qgs3DUtils::captureSceneImage( engine, &scene );
+    img.save( fileName, fileFormat.toLocal8Bit().data() );
+    emit savedAsImage( fileName );
```

Comment:
Hmm, this means that a copy of the whole scene needs to be created, and possibly many chunks need to be loaded - that could possibly take many seconds, and consume (temporarily) a lot of GPU memory. Are we sure we want to do it? Why not just use the current 3D engine and capture the image?